### PR TITLE
[BUGFIX beta] Fix Brittle Promise Usage

### PR DESCRIPTION
**Pull request by bmac**
This pr attempts to fix the issues identified in https://github.com/emberjs/data/issues/4488 by wrapping the ajax success/error code in a try/catch.

cc @stefanpenner @pangratz for review.

### CODE REVIEW

File: addon/adapters/rest.js [hunk 1]
```diff
@@ -1454,6 +1405,50 @@ if (isEnabled('ds-improved-ajax')) {
 
 }
 
+function ajaxSuccess(adapter, jqXHR, payload, requestData, resolve, reject) {
```
Comment (stefanpenner) [inline addon/adapters/rest.js]: lets not pass reject/resolve here, instead simply `return` or `throw` (or `return Promise.reject()`)

**Comment by bmac** [comment]
@stefanpenner I've updated the pr do you mind giving it another review?

File: addon/adapters/rest.js [hunk 2]
```diff
@@ -1001,52 +1003,30 @@ var RESTAdapter = Adapter.extend(BuildURLMixin, {
...
-          parseResponseHeaders(jqXHR.getAllResponseHeaders()),
-          payload,
-          requestData
-        );
-
-        if (response && response.isAdapterError) {
-          Ember.run.join(null, reject, response);
-        } else {
-          Ember.run.join(null, resolve, response);
+        var successValue;
```
Comment (pangratz) [inline addon/adapters/rest.js]: This avriable can be moved inside the `try catch` block.

File: addon/adapters/rest.js (hunk 1 above)
Comment (pangratz) [inline addon/adapters/rest.js]: These unused params can be removed...

**Comment by pangratz** [comment]
LGTM 👍

File: addon/adapters/rest.js [hunk 3]
```diff
@@ -25,6 +25,8 @@ const {
   get
 } = Ember;
 
+const Promise = Ember.RSVP.Promise;
```
Comment (locks) [inline addon/adapters/rest.js]: could you import RSVP directly here, instead of using the Ember namespaced one?

Comment (fivetanley) [inline addon/adapters/rest.js]: I don't think we have support for that for the globals build

**Pull request merged**